Makefile

```diff
@@ -286,6 +322,50 @@ _build/stage2/lib/template-hsc.h: utils/hsc2hs/data/template-hsc.h
 .PHONY: stage2
 stage2: $(addprefix _build/stage2/bin/,$(STAGE2_EXECUTABLES)) _build/stage2/lib/settings _build/stage2/lib/package.conf.d/package.cache _build/stage2/lib/template-hsc.h
 
+.PHONY: stage2-javascript-unknown-ghcjs
+stage2-javascript-unknown-ghcjs: javascript-unknown-ghcjs-libs _build/stage2/lib/targets/javascript-unknown-ghcjs/lib/package.conf.d/package.cache
+
+_build/stage2/lib/targets/javascript-unknown-ghcjs/lib/settings: _build/stage1/bin/ghc-toolchain-bin
+	@mkdir -p $(@D)
+	$(call run_and_log, emconfigure _build/stage1/bin/ghc-toolchain-bin --triple javascript-unknown-ghcjs --output-settings -o $@ --cc $(EMCC) --cxx $(EMCXX) --ar $(EMAR) --ranlib $(EMRANLIB))
+
+_build/stage2/bin/javascript-unknown-ghcjs-ghc-pkg: _build/stage2/bin/ghc-pkg
+	@mkdir -p $(@D)
+	ln -sf ghc-pkg $@
+
+_build/stage2/bin/javascript-unknown-ghcjs-ghc: _build/stage2/bin/ghc
+	@mkdir -p $(@D)
+	ln -sf ghc $@
+
+_build/stage2/lib/targets/javascript-unknown-ghcjs/lib/package.conf.d:
+	@mkdir -p $@
+
+_build/stage2/lib/targets/javascript-unknown-ghcjs/lib/package.conf.d/package.cache: _build/stage2/bin/javascript-unknown-ghcjs-ghc-pkg _build/stage2/lib/targets/javascript-unknown-ghcjs/lib/settings javascript-unknown-ghcjs-libs
+	@mkdir -p $(@D)
+	@rm -rf $(@D)/*
+	cp -rfp _build/stage2/javascript-unknown-ghcjs/packagedb/host/*/* $(@D)
+	_build/stage2/bin/javascript-unknown-ghcjs-ghc-pkg recache
+
+_build/stage2/lib/targets/javascript-unknown-ghcjs/bin/unlit: _build/stage2/bin/unlit
+	@mkdir -p $(@D)
+	cp -rfp $< $@
+
+.PHONY: javascript-unknown-ghcjs-libs
+javascript-unknown-ghcjs-libs: private TARGET=javascript-unknown-ghcjs
+javascript-unknown-ghcjs-libs: private GHC=$(abspath _build/stage2/bin/javascript-unknown-ghcjs-ghc)
+javascript-unknown-ghcjs-libs: private GHC2=$(abspath _build/stage2/bin/ghc)
+javascript-unknown-ghcjs-libs: private STAGE=stage2
+javascript-unknown-ghcjs-libs: private CC=emcc
+javascript-unknown-ghcjs-libs: _build/stage2/bin/javascript-unknown-ghcjs-ghc-pkg _build/stage2/bin/javascript-unknown-ghcjs-ghc _build/stage2/lib/targets/javascript-unknown-ghcjs/lib/settings _build/stage2/lib/targets/javascript-unknown-ghcjs/bin/unlit _build/stage2/lib/targets/javascript-unknown-ghcjs/lib/package.conf.d
+	# Force cabal to replan
+	rm -rf _build/stage2/javascript-unknown-ghcjs/cache
+	$(call run_and_log, HADRIAN_SETTINGS='$(HADRIAN_SETTINGS)' \
+		PATH=$(PWD)/_build/stage2/bin:$(PATH) \
+		$(CABAL_BUILD) -W $(GHC2) --happy-options="--template=$(abspath _build/stage2/src/happy-lib-2.1.5/data/)" rts:nonthreaded-nodebug )
```

Comment:
Do you remember what the underlying reason for the happy-lib-... path thing was? Can we add a comment to remember why this was needed?